pkg/controllers/mcs/mcs_controller.go

```diff
@@ -0,0 +1,380 @@
+/*
+Copyright 2023 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mcs
```

Comment:
To differentiate the current mcs controller, I believe it would be best to move it to a different directory.

Reply:
cc @RainbowMango for better suggestions.

Options:
1. mcsnative

Reply:
It's ok to put the controller in this package. the user-facing controller name is defined at https://github.com/karmada-io/karmada/pull/4323/files#diff-a111978a679a9f03e49c67a43ecb913299d153e59ce48d2d30f5029f27ab601aR227.

Reply:
I think we should also put it in another folder for easier code analysis and reading.